cpp/src/groupby/sort/group_tdigest.cu

```diff
@@ -0,0 +1,849 @@
+  auto weight = tdigest.child(cudf::detail::tdigest::weight_column_index);
+
+  // first step is to merge all the tdigests in each group. at the moment the only way to
+  // make this work is to retrieve the group sizes (via group_offsets) and the individual digest
+  // sizes (via input.offsets()) to the gpu and do the merges.  The scale problem is that while the
+  // size of each group will likely be small (size of each group will typically map to # of batches
+  // the input data was chopped into for tdigest generation), the -number- of groups can be
+  // arbitrarily large.
+  //
+  // thrust::merge and thrust::merge_by_key don't provide what we need.  What we would need is an
+  // algorithm like a super-merge that takes two layers of keys: one which identifies the outer
+  // grouping of tdigests, and one which identifies the inner groupings of the tdigests within the
+  // outer groups.
+
+  // bring group offsets back to the host
+  std::vector<size_type> h_outer_offsets(group_offsets.size());
+  cudaMemcpyAsync(h_outer_offsets.data(),
+                  group_offsets.data(),
+                  sizeof(size_type) * group_offsets.size(),
+                  cudaMemcpyDeviceToHost,
+                  stream);
+
+  // bring tdigest offsets back to the host
+  auto tdigest_offsets = lcv.offsets();
+  std::vector<size_type> h_inner_offsets(tdigest_offsets.size());
+  cudaMemcpyAsync(h_inner_offsets.data(),
+                  tdigest_offsets.begin<size_type>(),
+                  sizeof(size_type) * tdigest_offsets.size(),
+                  cudaMemcpyDeviceToHost,
+                  stream);
+
+  stream.synchronize();
+
+  // extract all means and weights into a table
+  cudf::table_view tdigests_unsliced({mean, weight});
+
+  // generate the merged (but not yet compressed) tdigests for each group.
+  std::vector<std::unique_ptr<table>> tdigests;
+  tdigests.reserve(num_groups);
+  auto iter = thrust::make_counting_iterator(0);
+  std::transform(iter, iter + num_groups, std::back_inserter(tdigests), [&](int outer_index) {
+    // the range of tdigests in this group
+    auto const tdigest_start = h_outer_offsets[outer_index];
+    auto const tdigest_end   = h_outer_offsets[outer_index + 1];
+    auto const num_tdigests  = tdigest_end - tdigest_start;
+
+    // slice each tdigest from the input
+    std::vector<table_view> unmerged_tdigests;
+    unmerged_tdigests.reserve(num_tdigests);
+    auto offset_iter = thrust::make_counting_iterator(tdigest_start);
+    std::transform(offset_iter,
+                   offset_iter + num_tdigests,
+                   std::back_inserter(unmerged_tdigests),
+                   [&](int inner_index) {
+                     std::vector<size_type> indices{h_inner_offsets[inner_index],
+                                                    h_inner_offsets[inner_index + 1]};
+                     return cudf::detail::slice(tdigests_unsliced, indices, stream);
+                   });
```

Comment:
```suggestion
   auto offset_iter = std::next(h_inner_offsets.begin(), tdigest_start);
    std::transform(offset_iter,
                   offset_iter + num_tdigests,
                   std::next(offset_iter),
                   std::back_inserter(unmerged_tdigests),
                   [&](auto start, auto end) {
                     return cudf::detail::slice(tdigests_unsliced, {start,end}, stream);
                   });
```

Reply:
Can't.  This is a table slice.

Reply:
:( 

Well, should still be able to do this;

```
   auto offset_iter = std::next(h_inner_offsets.begin(), tdigest_start);
    std::transform(offset_iter,
                   offset_iter + num_tdigests,
                   std::next(offset_iter),
                   std::back_inserter(unmerged_tdigests),
                   [&](auto start, auto end) {
                     return cudf::detail::slice(tdigests_unsliced, {start,end}, stream);
                   });